src/extensions/idle.rs

```diff
@@ -226,13 +226,9 @@ impl<'a, T: Read + Write + 'a> FallibleIterator for IdleIterator<'a, T> {
             if let Ok(u) = self.handle.session.unsolicited_responses.try_recv() {
                 return Ok(Some(u));
             }
-
+            self.buffer.clear();
```

Comment:
This doesn't seem quite right? What if the first `readline` only gives you a partial response and you need to read more to get and parse the full response?

Reply:
parse_idle() parses until the buffer is empty, so it shouldn't happen.

Reply:
The issue is if you get _half_ of a valid response, and you need to read in the other half before parsing can get the next item out.

Reply:
Oh, right, there's a problem here.